### ISSUE TITLE
Use OPT_CALLBACK instead of OPTION_CALLBACK

### ISSUE THREAD

**Issue by Denton-L** [gitgitgadget/git#621]
In the codebase, we have many instances of `OPTION_CALLBACK` used directly in option arrays. It would be nice to convert these to use the `OPT_CALLBACK` or `OPT_CALLBACK_F` macros as applicable.

All instances that need converting can be trivially found using `git grep OPTION_CALLBACK`.

Here is an example of one conversion:

```diff
diff --git a/builtin/push.c b/builtin/push.c
index 6dbf0f0bb7..64b0295a24 100644
--- a/builtin/push.c
+++ b/builtin/push.c
@@ -552,9 +552,9 @@ int cmd_push(int argc, const char **argv, const char *prefix)
                  0, CAS_OPT_NAME, &cas, N_("<refname>:<expect>"),
                  N_("require old value of ref to be at this value"),
                  PARSE_OPT_OPTARG | PARSE_OPT_LITERAL_ARGHELP, parseopt_push_cas_option },
-               { OPTION_CALLBACK, 0, "recurse-submodules", &recurse_submodules, "(check|on-demand|no)",
+               OPT_CALLBACK_F(0, "recurse-submodules", &recurse_submodules, "(check|on-demand|no)",
                        N_("control recursive pushing of submodules"),
-                       PARSE_OPT_OPTARG, option_parse_recurse_submodules },
+                       PARSE_OPT_OPTARG, option_parse_recurse_submodules),
                OPT_BOOL_F( 0 , "thin", &thin, N_("use thin pack"), PARSE_OPT_NOCOMPLETE),
                OPT_STRING( 0 , "receive-pack", &receivepack, "receive-pack", N_("receive pack program")),
                OPT_STRING( 0 , "exec", &receivepack, "receive-pack", N_("receive pack program")),
```

**Comment by Denton-L** [comment]
Patch here: https://lore.kernel.org/git/20200428084543.GC2381876@coredump.intra.peff.net/T/#m255371eb8c9346e5276557c6a62144cca46cab1b

**Comment by Denton-L** [comment]
Fixed in 66527162008d2792c0000096dd7a41d6d5797df3.
